### PR TITLE
add timeout for apiserver registry store 

### PR DESCRIPTION
**Pull request by DylanBLE**
**What this PR does / why we need it**:
Enable etcd client balancing for apiserver. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #https://github.com/kubernetes/kubernetes/issues/69621

**Special notes for your reviewer**:

**Release note**:

```release-note
add timeout for apiserver registry store
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go [hunk 1]
```diff
@@ -280,6 +280,7 @@ func (e *Store) GetExportStrategy() rest.RESTExportStrategy {
 // List returns a list of items matching labels and field according to the
 // store's PredicateFunc.
 func (e *Store) List(ctx context.Context, options *metainternalversion.ListOptions) (runtime.Object, error) {
+	ctx, _ = context.WithTimeout(ctx, 10*time.Second)
```
Comment (wgliang) [inline staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go]: But why the time is 10s?

Comment (DylanBLE) [reply staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go]: It's an example of showing timeout context works. If this solution is accepted, I can change it to a command line flag or a more appropriate value.

Comment (wgliang) [reply staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go]: I think this timeout size setting needs to be supported by the basis, not an experimental value.
:)

**Comment by lavalamp** [comment]
@jingyih is looking at wiring up the correct timeout here; we already have a timeout set at other layers of the stack that's just not threaded through.

A hard limit of 10s (or any other particular value) isn't feasible given we have real values and they depend on what the client asks for.

**Comment by liggitt** [comment]
the parent issue was reported against 1.9.x. https://github.com/kubernetes/kubernetes/commit/31ff8c6b9a08d1168502e423bdd3fdbe4f2f729b added a keep-alive timeout in 1.10, which should cover this

**Comment by DylanBLE** [comment]
31ff8c6 fixed the issue.

**Pull request closed without merging**